core/src/main/java/fr/sncf/osrd/envelope_sim_infra/ertms/etcs/BrakingCurves.java

```diff
@@ -0,0 +1,155 @@
+package fr.sncf.osrd.envelope_sim_infra.ertms.etcs;
+
+import static fr.sncf.osrd.envelope.part.constraints.EnvelopePartConstraintType.CEILING;
+import static fr.sncf.osrd.envelope.part.constraints.EnvelopePartConstraintType.FLOOR;
+import static fr.sncf.osrd.envelope_sim.EnvelopeSimContext.UseCase.*;
+import static fr.sncf.osrd.envelope_sim_infra.ertms.etcs.BrakingCurves.BrakingCurveType.*;
+
+import fr.sncf.osrd.envelope.Envelope;
+import fr.sncf.osrd.envelope.EnvelopeCursor;
+import fr.sncf.osrd.envelope.part.ConstrainedEnvelopePartBuilder;
+import fr.sncf.osrd.envelope.part.EnvelopePart;
+import fr.sncf.osrd.envelope.part.EnvelopePartBuilder;
+import fr.sncf.osrd.envelope.part.constraints.EnvelopeConstraint;
+import fr.sncf.osrd.envelope.part.constraints.SpeedConstraint;
+import fr.sncf.osrd.envelope_sim.EnvelopeProfile;
+import fr.sncf.osrd.envelope_sim.EnvelopeSimContext;
+import fr.sncf.osrd.envelope_sim.overlays.EnvelopeDeceleration;
+import fr.sncf.osrd.envelope_sim.pipelines.MaxSpeedEnvelope;
+import fr.sncf.osrd.envelope_sim_infra.EnvelopeTrainPath;
+import fr.sncf.osrd.infra_state.api.TrainPath;
+import fr.sncf.osrd.train.RollingStock;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BrakingCurves {
+
+    public enum BrakingCurveType {
+        EBD,
+        EBI,
+        SBD,
+        SBI1,
+        SBI2,
+        GUI,
+        PS,
+        IND
+    }
+
+    /** Computes the ETCS braking curves from a path, a schedule, a time step, and a MRSP envelope */
+    public static List<EnvelopePart> from(
```

Comment:
I like how this file has a single public method and the rest is divided in small private methods.

But I don't think a list of envelope parts is the most convenient way to handle the result. Why not something like a `Multimap<BrakingCurveType, EnvelopePart>`? What will we need when iterating over the result?